src/olympia/reviews/tests/test_views.py

```diff
@@ -730,3 +730,200 @@ def test_detail_deleted(self):
 
         response = self.client.get(self.url)
         assert response.status_code == 404
+
+
+class TestReviewViewSetPost(TestCase):
+    client_class = APITestClient
+
+    def setUp(self):
+        self.addon = addon_factory(
+            guid=generate_addon_guid(), name=u'My Addôn', slug='my-addon')
+        self.url = reverse(
+            'addon-review-list', kwargs={'addon_pk': self.addon.pk})
+
+    def test_post_anonymous(self):
+        response = self.client.post(self.url, {
+            'body': u'test bodyé', 'title': None, 'rating': 5})
+        assert response.status_code == 401
+
+    def test_post_no_version(self):
+        self.user = user_factory()
+        self.client.login_api(self.user)
+        assert not Review.objects.exists()
+        response = self.client.post(self.url, {
+            'body': u'test bodyé', 'title': None, 'rating': 5})
+        assert response.status_code == 400
+
+    def test_post_version_string(self):
+        self.user = user_factory()
+        self.client.login_api(self.user)
+        assert not Review.objects.exists()
+        response = self.client.post(self.url, {
+            'body': u'test bodyé', 'title': None, 'rating': 5,
+            'version': self.addon.current_version.version})
+        assert response.status_code == 400
+
+    def test_post_logged_in(self):
+        self.user = user_factory()
+        self.client.login_api(self.user)
+        assert not Review.objects.exists()
+        response = self.client.post(self.url, {
+            'body': u'test bodyé', 'title': u'blahé', 'rating': 5,
+            'version': self.addon.current_version.pk})
+        assert response.status_code == 201
+        data = json.loads(response.content)
+        review = Review.objects.latest('pk')
+        assert review.pk == data['id']
+        assert unicode(review.body) == data['body'] == u'test bodyé'
+        assert review.rating == data['rating'] == 5
+        assert review.user == self.user
+        assert unicode(review.title) == data['title'] == u'blahé'
+        assert review.reply_to is None
+        assert review.addon == self.addon
+        assert review.version == self.addon.current_version
+        assert data['version'] == review.version.version
+
+    def test_post_rating_float(self):
+        self.user = user_factory()
+        self.client.login_api(self.user)
+        assert not Review.objects.exists()
+        response = self.client.post(self.url, {
+            'body': u'test bodyé', 'title': None, 'rating': 4.5,
+            'version': self.addon.current_version.pk})
+        assert response.status_code == 400
+
```

Comment:
Can we verify the correct error message or at least type here to make sure that the validation error we receive is because it's a float and maybe not because we decided to make `title` required sometime in the future?